src/CifFile_module.py

```diff
@@ -3077,16 +3077,15 @@ def validate_item_esd(self,item_name,item_value):
         return {"result":True}
 
     def validate_item_esd_ddlm(self,item_name,item_value):
-        if self[item_name].get('self.primitive_type') not in \
+        if self[item_name].get(self.primitive_type) not in \
         ['Count','Index','Integer','Real','Imag','Complex','Binary','Hexadecimal','Octal']:
             return {"result":None}
         can_esd = True
         if self[item_name].get('_type.purpose') != 'Measurand':
             can_esd = False
         item_values = listify(item_value)
```

Comment:
validate_item_esd just straight returns `{"result":True}` if `can_esd` is `True` here. Is that possible here, too?

Reply:
The logic is that a `Measurand` *must* have an su. In DDL1 I think it was more that it *could* have an su.

This whole method needs a complete rewrite, as the SU could be stored in a separate data name and so having no SU appended is not an issue.

Reply:
OK. I was afraid of that. I'll see what I can come up with.